File: android/app/src/main/java/com/ichigo/app/ui/components/DesignComponents.kt
package com.ichigo.app.ui.components

import androidx.compose.animation.core.animateDpAsState
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicTextField
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.VolumeUp
import androidx.compose.material.icons.filled.Cancel
import androidx.compose.material.icons.filled.ChevronLeft
import androidx.compose.material.icons.filled.DarkMode
import androidx.compose.material.icons.filled.Inbox
import androidx.compose.material.icons.filled.LightMode
import androidx.compose.material.icons.filled.Search
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.clipToBounds
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.ichigo.app.ui.theme.Dimens
import com.ichigo.app.ui.theme.IchigoPalette
import com.ichigo.app.ui.theme.IchigoTheme
import com.ichigo.app.ui.theme.Wt
import com.ichigo.app.ui.theme.rounded
import com.ichigo.app.ui.theme.softShadow
import com.ichigo.app.util.LocalSpeech

/** A rounded surface card with the app's soft shadow (the common list/detail card). */
fun Modifier.ichigoCard(
    surface: Color,
    shadow: Color,
    radius: androidx.compose.ui.unit.Dp = Dimens.CardRadius,
    shadowRadius: androidx.compose.ui.unit.Dp = 9.dp,
    shadowY: androidx.compose.ui.unit.Dp = 6.dp,
): Modifier = this
    .softShadow(shadow, shadowRadius, radius, offsetY = shadowY)
    .clip(RoundedCornerShape(radius))
    .background(surface)

/** Pinned screen header with a circular back button + heavy title (Swift `ScreenHeader`). */
@Composable
fun ScreenHeader(title: String, onBack: () -> Unit) {
    val c = IchigoTheme.colors
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 20.dp)
            .padding(top = 8.dp, bottom = 4.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(
            modifier = Modifier
                .size(44.dp)
                .softShadow(c.cardShadow, 6.dp, 22.dp, offsetY = 2.dp)
                .clip(CircleShape)
                .background(c.surface)
                .clickable(onClick = onBack),
            contentAlignment = Alignment.Center,
        ) {
            Icon(Icons.Filled.ChevronLeft, contentDescription = "Kembali", tint = c.primaryText)
        }
        Spacer(Modifier.width(14.dp))
        Text(
            title,
            style = rounded(24, Wt.Heavy),
            color = c.primaryText,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
        )
    }
}

/** Rounded pill search field (Swift `SearchField`). */
@Composable
fun SearchField(
    value: String,
    onValueChange: (String) -> Unit,
    placeholder: String,
    modifier: Modifier = Modifier,
) {
    val c = IchigoTheme.colors
    Row(
        modifier = modifier
            .fillMaxWidth()
            .height(54.dp)
            .softShadow(c.cardShadow, 8.dp, 27.dp, offsetY = 3.dp)
            .clip(RoundedCornerShape(27.dp))
            .background(c.surface)
            .padding(horizontal = 20.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(Icons.Filled.Search, contentDescription = null, tint = c.secondaryText)
        Spacer(Modifier.width(12.dp))
        Box(Modifier.weight(1f)) {
            if (value.isEmpty()) {
                Text(placeholder, style = rounded(16, Wt.Medium), color = IchigoPalette.Placeholder)
            }
            BasicTextField(
                value = value,
                onValueChange = onValueChange,
                singleLine = true,
                textStyle = rounded(16, Wt.Medium).merge(TextStyle(color = c.primaryText)),
                cursorBrush = androidx.compose.ui.graphics.SolidColor(IchigoPalette.Accent),
                keyboardOptions = androidx.compose.foundation.text.KeyboardOptions(imeAction = ImeAction.Search),
                modifier = Modifier.fillMaxWidth(),
            )
        }
        if (value.isNotEmpty()) {
            Spacer(Modifier.width(8.dp))
            Icon(
                Icons.Filled.Cancel,
                contentDescription = "Hapus pencarian",
                tint = c.secondaryText,
                modifier = Modifier.size(18.dp).clip(CircleShape).clickable { onValueChange("") },
            )
        }
    }
}

/** Blue gradient hero card with two decorative circles (Swift `DetailHeroCard`). */
@Composable
fun DetailHeroCard(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .softShadow(IchigoPalette.Blue.copy(alpha = 0.28f), 16.dp, Dimens.HeroRadius, offsetY = 8.dp)
            .clip(RoundedCornerShape(Dimens.HeroRadius))
            .background(Brush.linearGradient(listOf(IchigoPalette.BlueLight, IchigoPalette.Blue)))
            .clipToBounds(),
    ) {
        // Decorative corner circles.
        Box(
            Modifier
                .size(190.dp)
                .offset(x = 230.dp, y = (-70).dp)
                .clip(CircleShape)
                .background(Color.White.copy(alpha = 0.12f)),
        )
        Box(
            Modifier
                .size(150.dp)
                .offset(x = (-60).dp, y = 120.dp)
                .clip(CircleShape)
                .background(Color.White.copy(alpha = 0.10f)),
        )
        Box(Modifier.padding(20.dp)) { content() }
    }
}

/** Solid white badge over the hero card, e.g. "JLPT N5" (Swift `HeroBadge`). */
@Composable
fun HeroBadge(text: String) {
    Text(
        text,
        style = rounded(11, Wt.Bold),
        color = IchigoPalette.Blue,
        modifier = Modifier
            .clip(RoundedCornerShape(50))
            .background(Color.White)
            .padding(horizontal = 12.dp, vertical = 6.dp),
    )
}

/** Translucent pill over the hero card (Swift `HeroPill`). */
@Composable
fun HeroPill(text: String) {
    Text(
        text,
        style = rounded(11, Wt.Semibold),
        color = Color.White,
        modifier = Modifier
            .clip(RoundedCornerShape(50))
            .background(Color.White.copy(alpha = 0.22f))
            .padding(horizontal = 12.dp, vertical = 6.dp),
    )
}

/** Round translucent speak button over the hero card (Swift `HeroSpeakButton`). */
@Composable
fun HeroSpeakButton(text: String) {
    val speech = LocalSpeech.current
    Box(
        Modifier
            .size(38.dp)
            .clip(CircleShape)
            .background(Color.White.copy(alpha = 0.22f))
            .clickable { speech.speak(text) },
        contentAlignment = Alignment.Center,
    ) {
        Icon(Icons.AutoMirrored.Filled.VolumeUp, contentDescription = "Dengarkan pelafalan", tint = Color.White, modifier = Modifier.size(18.dp))
    }
}

/** White section card with a small gradient-tinted icon (Swift `DetailSectionCard`). */
@Composable
fun DetailSectionCard(
    title: String,
    icon: ImageVector,
    tint: Color = IchigoPalette.Blue,
    trailing: @Composable (() -> Unit)? = null,
    content: @Composable () -> Unit,
) {
    val c = IchigoTheme.colors
    Column(
        Modifier
            .fillMaxWidth()
            .ichigoCard(c.surface, c.cardShadow, shadowRadius = 8.dp, shadowY = 3.dp)
            .padding(16.dp),
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Box(
                Modifier
                    .size(30.dp)
                    .clip(RoundedCornerShape(10.dp))
                    .background(tint.copy(alpha = 0.12f)),
                contentAlignment = Alignment.Center,
            ) {
                Icon(icon, contentDescription = null, tint = tint, modifier = Modifier.size(15.dp))
            }
            Spacer(Modifier.width(10.dp))
            Text(title, style = rounded(17, Wt.Bold), color = c.primaryText)
            Spacer(Modifier.weight(1f))
            if (trailing != null) trailing()
        }
        Spacer(Modifier.height(14.dp))
        content()
    }
}

/** Horizontally scrolling filter chips (Swift `FilterChipRow`). */
@Composable
fun FilterChipRow(filters: List<String>, selected: String, onSelect: (String) -> Unit) {
    val c = IchigoTheme.colors
    LazyRow(
        contentPadding = PaddingValues(horizontal = 20.dp, vertical = 4.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        items(filters) { filter ->
            val isSelected = filter == selected
            Box(
                Modifier
                    .height(46.dp)
                    .softShadow(c.cardShadow, 6.dp, 23.dp, offsetY = 2.dp)
                    .clip(RoundedCornerShape(23.dp))
                    .background(if (isSelected) IchigoPalette.Accent else c.surface)
                    .clickable { onSelect(filter) }
                    .padding(horizontal = 22.dp),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    filter,
                    style = rounded(15, Wt.Bold),
                    color = if (isSelected) Color.White else c.secondaryText,
                )
            }
        }
    }
}

/** Sun/moon slide toggle for the theme (Swift `ThemeSlideToggle`). */
@Composable
fun ThemeSlideToggle(isDark: Boolean, onToggle: (Boolean) -> Unit) {
    val trackWidth = 78.dp
    val trackHeight = 40.dp
    val knobSize = 32.dp
    val inset = 4.dp
    val travel = (trackWidth - knobSize) / 2 - inset
    val offsetX by animateDpAsState(if (isDark) travel else -travel, spring(dampingRatio = 0.72f, stiffness = 900f), label = "knob")

    Box(
        Modifier
            .size(trackWidth, trackHeight)
            .clip(RoundedCornerShape(50))
            .background(
                Brush.horizontalGradient(
                    if (isDark) listOf(IchigoPalette.IndigoDeep, IchigoPalette.Navy)
                    else listOf(IchigoPalette.BlueLight, IchigoPalette.Blue),
                ),
            )
            .clickable { onToggle(!isDark) },
        contentAlignment = Alignment.Center,
    ) {
        Row(
            Modifier.fillMaxWidth().padding(horizontal = 11.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Icon(Icons.Filled.LightMode, null, tint = Color.White.copy(alpha = if (isDark) 0.45f else 0f), modifier = Modifier.size(15.dp))
            Icon(Icons.Filled.DarkMode, null, tint = Color.White.copy(alpha = if (isDark) 0f else 0.5f), modifier = Modifier.size(15.dp))
        }
        Box(
            Modifier
                .offset(x = offsetX)
                .size(knobSize)
                .clip(CircleShape)
                .background(Color.White),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                if (isDark) Icons.Filled.DarkMode else Icons.Filled.LightMode,
                contentDescription = "Mode tampilan",
                tint = if (isDark) IchigoPalette.IndigoDeep else IchigoPalette.Caution,
                modifier = Modifier.size(16.dp),
            )
        }
    }
}

/** Empty/error/loading states (Swift `EmptyStateView` / `ErrorStateView`). */
@Composable
fun EmptyState(title: String, subtitle: String, icon: ImageVector = Icons.Filled.Inbox, modifier: Modifier = Modifier) {
    val c = IchigoTheme.colors
    Column(
        modifier = modifier.padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        Icon(icon, null, tint = c.secondaryText, modifier = Modifier.size(42.dp))
        Text(title, style = rounded(16, Wt.Bold), color = c.primaryText)
        Text(subtitle, style = rounded(13), color = c.secondaryText, textAlign = TextAlign.Center)
    }
}

@Composable
fun ErrorState(message: String, modifier: Modifier = Modifier) =
    EmptyState("Gagal memuat data", message, Icons.Filled.Warning, modifier)

File: ios/Sources/AppFeature/DesignComponents.swift
import SwiftUI

// MARK: - Screen Header

/// Pinned screen header: a circular back button next to a bold rounded title.
/// Replaces the system navigation bar so the title stays put while content
/// scrolls underneath it.
struct ScreenHeader: View {
    let title: String
    var onBack: (() -> Void)?

    @Environment(\.dismiss) private var dismiss
    @Environment(\.colorScheme) private var scheme

    var body: some View {
        HStack(spacing: 14) {
            Button {
                if let onBack { onBack() } else { dismiss() }
            } label: {
                Image(systemName: "chevron.left")
                    .font(AppTheme.rounded(17, .bold))
                    .foregroundStyle(AppTheme.primaryText(scheme))
                    .frame(width: 44, height: 44)
                    .background(AppTheme.surface(scheme))
                    .clipShape(Circle())
                    .shadow(color: AppTheme.softShadow(scheme), radius: 6, x: 0, y: 2)
            }
            .buttonStyle(.plain)
            .accessibilityLabel("Kembali")

            Text(title)
                .font(AppTheme.rounded(24, .heavy))
                .foregroundStyle(AppTheme.primaryText(scheme))
                .lineLimit(1)
                .minimumScaleFactor(0.7)

            Spacer(minLength: 0)
        }
        .padding(.horizontal, 20)
        .padding(.top, 8)
        .padding(.bottom, 4)
    }
}

// MARK: - Search Field

/// Rounded pill search field used across the browsing screens.
struct SearchField: View {
    let placeholder: String
    @Binding var text: String

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        HStack(spacing: 12) {
            Image(systemName: "magnifyingglass")
                .font(AppTheme.rounded(17, .semibold))
                .foregroundStyle(AppTheme.secondaryText(scheme))

            TextField(placeholder, text: $text)
                .font(AppTheme.rounded(16, .medium))
                .textInputAutocapitalization(.never)
                .disableAutocorrection(true)

            if !text.isEmpty {
                Button {
                    text = ""
                } label: {
                    Image(systemName: "xmark.circle.fill")
                        .font(AppTheme.rounded(16))
                        .foregroundStyle(AppTheme.secondaryText(scheme))
                }
                .buttonStyle(.plain)
                .accessibilityLabel("Hapus pencarian")
            }
        }
        .padding(.horizontal, 20)
        .frame(height: 54)
        .background(AppTheme.surface(scheme))
        .clipShape(Capsule())
        .shadow(color: AppTheme.softShadow(scheme), radius: 8, x: 0, y: 3)
    }
}

// MARK: - Kartu Hero Biru

/// Kartu hero bergradien biru yang dipakai di puncak layar detail (Detail Kanji
/// dan Detail Tata Bahasa).
///
/// Dua lingkaran tembus pandang di sudutnya adalah hiasan dari desain; keduanya
/// dipotong mengikuti sudut kartu lewat `clipShape` sehingga tidak meluber.
struct DetailHeroCard<Content: View>: View {
    @ViewBuilder let content: Content

    var body: some View {
        content
            .padding(20)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(
                ZStack {
                    LinearGradient(
                        colors: [AppTheme.blueLight, AppTheme.blue],
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    )

                    // Hiasan sudut kanan atas dan kiri bawah.
                    GeometryReader { geo in
                        Circle()
                            .fill(Color.white.opacity(0.12))
                            .frame(width: 190, height: 190)
                            .offset(x: geo.size.width - 110, y: -70)

                        Circle()
                            .fill(Color.white.opacity(0.10))
                            .frame(width: 150, height: 150)
                            .offset(x: -60, y: geo.size.height - 80)
                    }
                }
            )
            .clipShape(RoundedRectangle(cornerRadius: AppTheme.heroRadius, style: .continuous))
            .shadow(color: AppTheme.blue.opacity(0.28), radius: 16, x: 0, y: 8)
    }
}

/// Lencana putih pekat di atas kartu hero, mis. "JLPT N5".
struct HeroBadge: View {
    let text: String

    var body: some View {
        Text(text)
            .font(AppTheme.rounded(11, .bold))
            .foregroundStyle(AppTheme.blue)
            .padding(.horizontal, 12)
            .padding(.vertical, 6)
            .background(Color.white)
            .clipShape(Capsule())
    }
}

/// Pil tembus pandang di atas kartu hero, untuk keterangan tambahan seperti
/// pola struktur atau kategori.
struct HeroPill: View {
    let text: String

    var body: some View {
        Text(text)
            .font(AppTheme.rounded(11, .semibold))
            .foregroundStyle(.white)
            .padding(.horizontal, 12)
            .padding(.vertical, 6)
            .background(Color.white.opacity(0.22))
            .clipShape(Capsule())
    }
}

/// Tombol suara bundar tembus pandang di atas kartu hero.
///
/// Tombol ini menerima teksnya langsung, bukan sebuah closure, karena
/// `AudioSpeechHelper` terikat main actor: closure yang disimpan sebagai
/// properti tidak mewarisi isolasi itu, sedangkan closure yang ditulis di dalam
/// `body` mewarisinya.
struct HeroSpeakButton: View {
    let text: String

    var body: some View {
        Button {
            AudioSpeechHelper.shared.speak(text)
        } label: {
            Image(systemName: "speaker.wave.2.fill")
                .font(AppTheme.rounded(15, .semibold))
                .foregroundStyle(.white)
                .frame(width: 38, height: 38)
                .background(Color.white.opacity(0.22))
                .clipShape(Circle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel("Dengarkan pelafalan \(text)")
    }
}

// MARK: - Bagian Berkartu

/// Kartu putih dengan judul dan ikon kecil bergradien lembut di kirinya, dipakai
/// untuk bagian-bagian di layar detail ("Arti", "Penggunaan", dan seterusnya).
struct DetailSectionCard<Content: View, Trailing: View>: View {
    let title: String
    let systemImage: String
    var tint: Color = AppTheme.blue
    @ViewBuilder let content: Content
    @ViewBuilder let trailing: Trailing

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        VStack(alignment: .leading, spacing: 14) {
            HStack(spacing: 10) {
                Image(systemName: systemImage)
                    .font(AppTheme.rounded(13, .semibold))
                    .foregroundStyle(tint)
                    .frame(width: 30, height: 30)
                    .background(tint.opacity(0.12))
                    .clipShape(RoundedRectangle(cornerRadius: 10, style: .continuous))

                Text(title)
                    .font(AppTheme.rounded(17, .bold))
                    .foregroundStyle(AppTheme.primaryText(scheme))

                Spacer(minLength: 8)

                trailing
            }

            content
        }
        .padding(16)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(AppTheme.surface(scheme))
        .clipShape(RoundedRectangle(cornerRadius: AppTheme.cardRadius, style: .continuous))
        .shadow(color: AppTheme.cardShadow(scheme), radius: 8, x: 0, y: 3)
    }
}

extension DetailSectionCard where Trailing == EmptyView {
    init(title: String, systemImage: String, tint: Color = AppTheme.blue, @ViewBuilder content: () -> Content) {
        self.init(title: title, systemImage: systemImage, tint: tint, content: content, trailing: { EmptyView() })
    }
}

// MARK: - Filter Chips

/// Horizontally scrolling pill filters. The selected pill is filled with the
/// accent colour; the rest sit on the raised surface colour.
struct FilterChipRow: View {
    let filters: [String]
    @Binding var selection: String

    @Environment(\.colorScheme) private var scheme

    var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: 12) {
                ForEach(filters, id: \.self) { filter in
                    let isSelected = filter == selection
                    Button {
                        selection = filter
                    } label: {
                        Text(filter)
                            .font(AppTheme.rounded(15, .bold))
                            .foregroundStyle(isSelected ? .white : AppTheme.secondaryText(scheme))
                            .padding(.horizontal, 22)
                            .frame(height: 46)
                            .background(isSelected ? AppTheme.accent : AppTheme.surface(scheme))
                            .clipShape(Capsule())
                            .shadow(color: AppTheme.softShadow(scheme), radius: 6, x: 0, y: 2)
                    }
                    .buttonStyle(.plain)
                    .accessibilityAddTraits(isSelected ? [.isSelected] : [])
                }
            }
            .padding(.horizontal, 20)
            .padding(.vertical, 4)
        }
    }
}

// MARK: - Sakelar Geser Tema

/// Sakelar geser terang/gelap. Geser atau ketuk ke kiri untuk terang, ke kanan
/// untuk gelap; knopnya menampilkan matahari saat terang dan bulan saat gelap.
///
/// Terikat pada `Bool` sederhana (mati = terang, hidup = gelap). Pemanggil yang
/// menerjemahkannya ke pilihan tersimpan, sehingga sakelar ini tetap murni soal
/// tampilan.
struct ThemeSlideToggle: View {
    @Binding var isDark: Bool

    private let trackWidth: CGFloat = 78
    private let trackHeight: CGFloat = 40
    private let knobSize: CGFloat = 32
    private let inset: CGFloat = 4

    /// Jarak geser knop dari sisi ke sisi.
    private var knobTravel: CGFloat { (trackWidth - knobSize) / 2 - inset }

    var body: some View {
        ZStack {
            Capsule()
                .fill(
                    LinearGradient(
                        colors: isDark
                        ? [AppTheme.indigoDeep, AppTheme.navy]
                        : [AppTheme.blueLight, AppTheme.blue],
                        startPoint: .leading,
                        endPoint: .trailing
                    )
                )

            // Lambang matahari dan bulan sebagai penanda arah di balik knop.
            HStack {
                Image(systemName: "sun.max.fill")
                    .opacity(isDark ? 0.45 : 0)
                Spacer()
                Image(systemName: "moon.fill")
                    .opacity(isDark ? 0 : 0.5)
            }
            .font(AppTheme.rounded(13, .semibold))
            .foregroundStyle(.white)
            .padding(.horizontal, 11)

            Circle()
                .fill(Color.white)
                .frame(width: knobSize, height: knobSize)
                .overlay(
                    Image(systemName: isDark ? "moon.fill" : "sun.max.fill")
                        .font(AppTheme.rounded(14, .bold))
                        .foregroundStyle(isDark ? AppTheme.indigoDeep : AppTheme.caution)
                )
                .shadow(color: Color.black.opacity(0.18), radius: 4, x: 0, y: 2)
                .offset(x: isDark ? knobTravel : -knobTravel)
        }
        .frame(width: trackWidth, height: trackHeight)
        .contentShape(Capsule())
        .onTapGesture {
            withAnimation(.spring(response: 0.32, dampingFraction: 0.72)) {
                isDark.toggle()
            }
        }
        .gesture(
            DragGesture(minimumDistance: 6)
                .onEnded { value in
                    // Geser ke kanan → gelap, ke kiri → terang.
                    if value.translation.width > 10, !isDark {
                        withAnimation(.spring(response: 0.32, dampingFraction: 0.72)) { isDark = true }
                    } else if value.translation.width < -10, isDark {
                        withAnimation(.spring(response: 0.32, dampingFraction: 0.72)) { isDark = false }
                    }
                }
        )
        .accessibilityElement()
        .accessibilityLabel("Mode tampilan")
        .accessibilityValue(isDark ? "Gelap" : "Terang")
        .accessibilityAddTraits(.isButton)
        .accessibilityHint("Ketuk untuk beralih terang dan gelap")
    }
}
